vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/FieldValidationUtil.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2000-2019 Vaadin Ltd.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.vaadin.flow.component.textfield;
+
+import com.vaadin.flow.component.Component;
+import com.vaadin.flow.component.HasValidation;
+import com.vaadin.flow.component.page.PendingJavaScriptResult;
+
+/**
+ * Utility class for text field mixin web components to disable client
+ * side validation.
+ *
+ * @author Vaadin Ltd
+ */
+final class FieldValidationUtil {
+
+    private FieldValidationUtil() {
+        // utility class should not be instantiated
+    }
+
+    static void disableClientValidation(Component component) {
+        // if the component was already attached override the validate()
+        overrideClientValidation(component);
+
+        component.addAttachListener(e -> overrideClientValidation(component));
+    }
+
+    private static void overrideClientValidation(Component component) {
+        PendingJavaScriptResult javaScriptResult =
+                component.getElement()
+                        .executeJs("$0.validate = function () {return this.checkValidity();}",
```

Comment:
nit: There shouldn't be a need to use parameters here. `this` defaults to the element instance you call the `executeJs` on. Same thing in the other `executeJs` few lines below